### ISSUE TITLE
Question: Possible to cache prerenderResult?

### ISSUE THREAD

**Issue by Flood** [MarkPieszak/aspnetcore-angular2-universal#480]
We have seen that our page timeouts when there is "heavy" load (250 users) on the site. Do you think it is possible to add a cache to prerenderResult in HomeController?

And the key for the cache would be the actual route?

**Comment by rtozati** [comment]
Yes, you can cache prerender results. You can use in memory cache or Redis cache for example.

Follow a simple example using in memory cache:

`    using Microsoft.Extensions.Caching.Memory;

     ....

     private IMemoryCache _cache;

     ....

     public HomeController(IOptions<AppSettings> appSettings, IMemoryCache memoryCache)
    {
      _appSettings = appSettings;
      _cache = memoryCache;
    }

    [HttpGet]
    public async Task<IActionResult> Index()
    {
    ...
     // Prerender / Serialize application (with Universal)
     RenderToStringResult prerenderResult = null;

      // Look for cache key.
      if (!_cache.TryGetValue(unencodedAbsoluteUrl, out prerenderResult))
      {
        // Key not in cache, so get data.
        prerenderResult = await Prerenderer.RenderToString(
                "/",
                nodeServices,
                cancelToken,
                new JavaScriptModuleExport(applicationBasePath + "/ClientApp/dist/main-server"),
                unencodedAbsoluteUrl,
                unencodedPathAndQuery,
                transferData, // Our simplified Request object & any other CustommData you want to send!
                30000,
                Request.PathBase.ToString()
            );

        // Set cache options.
        var cacheEntryOptions = new MemoryCacheEntryOptions()
            // Keep in cache for this time, reset time if accessed.
            .SetSlidingExpiration(TimeSpan.FromSeconds(180));

        // Save data in cache.
        _cache.Set(unencodedAbsoluteUrl, prerenderResult, cacheEntryOptions);
      }
      ...
`

**Comment by Flood** [comment]
It works like a charm, thank you @rtozati!

**Comment by rtozati** [comment]
@Flood 

I saw that the usage of in memory cache does not scale. I implemented using Azure Redis Cache with way better performance.
